pygeos/test/test_strtree.py

```diff
@@ -629,6 +629,256 @@ def test_query_touches_polygons(poly_tree, geometry, expected):
+        # box overlaps envelope of 2 polygons but not covered by either
+        (box(0, 0, 1, 1), []),
+        # box covered by polygon
+        (box(0, 0, 0.5, 0.5), [0]),
+        # larger box intersects 3 polygons but not covered by any
+        (box(0, 0, 1.5, 1.5), []),
+        # buffer intersects 3 polygons but only within one
+        (pygeos.buffer(pygeos.points(3, 3), HALF_UNIT_DIAG), [3]),
+        # larger buffer overlaps 6 polygons (touches midpoints) but within none
+        (pygeos.buffer(pygeos.points(3, 3), 3 * HALF_UNIT_DIAG), []),
+        # envelope of points overlaps polygons, but points do not intersect
+        (pygeos.multipoints([[5, 7], [7, 5]]), []),
+        # only one point of multipoint within polygon
+        (pygeos.multipoints([[5, 7], [7, 7]]), []),
+        # both points in multipoint within polygon
+        (pygeos.multipoints([[5.25, 5.5], [5.25, 5.0]]), [5]),
+    ],
+)
+def test_query_covered_by_polygons(poly_tree, geometry, expected):
+    assert_array_equal(poly_tree.query(geometry, predicate="covered_by"), expected)
+
+
+### predicate == 'contains_properly'
+@pytest.mark.parametrize(
+    "geometry,expected",
+    [
+        # points do not intersect
+        (pygeos.points(0.5, 0.5), []),
+        # points intersect
+        (pygeos.points(1, 1), [1]),
+        # line contains every point that is not on its first or last coordinate
+        # these are on the "exterior" of the line
+        (pygeos.linestrings([[0, 0], [2, 2]]), [1]),
+        # slightly longer line contains multiple points
+        (pygeos.linestrings([[0.5, 0.5], [2.5, 2.5]]), [1, 2]),
+        # line intersects and contains one point
+        (pygeos.linestrings([[0, 2], [2, 0]]), [1]),
+        # box contains points (2 are at edges and not contained)
+        (box(3, 3, 6, 6), [4, 5]),
+        # envelope of buffer contains more points than within buffer
+        # due to diagonal distance
+        (pygeos.buffer(pygeos.points(3, 3), 1), [3]),
+        # envelope of buffer with 1/2 distance between points should intersect
+        # same points as envelope
+        (pygeos.buffer(pygeos.points(3, 3), 3 * HALF_UNIT_DIAG), [2, 3, 4]),
+        # multipoints intersect
+        (pygeos.multipoints([[5, 5], [7, 7]]), [5, 7]),
+        # envelope of points contains points, but points do not intersect
+        (pygeos.multipoints([[5, 7], [7, 5]]), []),
+        # only one point of multipoint intersects
+        (pygeos.multipoints([[5, 7], [7, 7]]), [7]),
+    ],
+)
+def test_query_contains_properly_points(tree, geometry, expected):
+    assert_array_equal(tree.query(geometry, predicate="contains_properly"), expected)
+
+
+@pytest.mark.parametrize(
+    "geometry,expected",
+    [
+        # None of the following conditions satisfy the relation for linestrings
+        # because they have no interior:
+        # "a contains b iff no points of b lie in the exterior of a, and at least one
```

Comment:
```suggestion
        # "a contains b if no points of b lie in the exterior of a, and at least one
```

Reply:
@jorisvandenbossche in this case, it was copied from the definition on the [wikipedia page](https://en.wikipedia.org/wiki/DE-9IM), with a special meaning of [`iff`](https://en.wikipedia.org/wiki/If_and_only_if) but perhaps the mathematical definition isn't as helpful as something that reads nicely in English?

Reply:
I had never heard of the abbreviation "iff" for "if and only if" ;) 
So personally I would say it's not that helpful here